pennylane/operation.py

```diff
@@ -1043,23 +1043,23 @@ def compare(self, other):
         >>> ob1.compare(ob2)
         False
         """
-        if isinstance(other, (Tensor, Observable)):
-            return other._obs_data() == self._obs_data()
         if isinstance(other, qml.Hamiltonian):
             return other.compare(self)
+        if isinstance(other, (Tensor, Observable)):
+            return other._obs_data() == self._obs_data()
```

Comment:
Order is changed, because Hamiltonian is now a subclass of Observable, so all cases would otherwise end in the first fork.